### PR TITLE
REF: Use numpy methods instead of np.array

### PR DESCRIPTION
**Pull request by mroeschke**
Minor, but avoid a call to `np.array` when we can use `np.full` or `reshape` or an existing op.

### CODE REVIEW

File: pandas/core/arrays/_ranges.py [hunk 1]
```diff
@@ -82,17 +84,7 @@ def generate_regular_range(
             "at least 'start' or 'end' should be specified if a 'period' is given."
         )
 
-    with np.errstate(over="raise"):
-        # If the range is sufficiently large, np.arange may overflow
-        #  and incorrectly return an empty array if not caught.
-        try:
-            values = np.arange(b, e, stride, dtype=np.int64)
-        except FloatingPointError:
```
Comment (jbrockmendel) [inline pandas/core/arrays/_ranges.py:90]: Do we have coverage here?

Comment (mroeschke) [reply pandas/core/arrays/_ranges.py:90]: I'm assuming there's coverage in the tests added in https://github.com/pandas-dev/pandas/pull/33498 where this code was added

**Review by rhshadrach — APPROVED** [review]
lgtm

**Comment by rhshadrach** [comment]
Thanks @mroeschke

**Pull request merged**